### ISSUE TITLE
Email and Picture from custom oAuth

### ISSUE THREAD

**Issue by ToniIltanen** [RocketChat/Rocket.Chat#7128]
Is there currently a way to map email and picture fields from custom oAuth through available configurations?
RC is receiving standard fields (email, picture) as part of the profile from the provider, but only the name -field is used in the rc user profile as mapped in the custom oAuth settings. 

The standard fields returned as the full profile (an example from our provider) : 
```javascript
claims() {
    return {
      address: {
        country: this.user.address.country,
        formatted: this.user.address.formatted,
        locality: this.user.address.locality,
        postal_code: this.user.address.postalCode,
        region: this.user.address.region,
        street_address: this.user.address.streetAdress,
      },
      birthdate: this.user.birthdate,
      email: this.user.email,
      email_verified: this.user.emailVerified,
      name: this.user.name,
      given_name: this.user.firstName,
      family_name: this.user.lastName,
      middle_name: "",
      locale: this.user.locale,
      phone_number: this.user.phone,
      picture: this.user.picture,
      sub: this.accountId,
      updated_at: 1454704946,
      zoneinfo: this.user.zoneInfo,
    };
  }
```
I could make the mappings in the custom oauth js for you guys, but data  -object in custom_oauth_server.js currently only assigns name, what are the other fields in the profile?

```javascript
const data = {
  serviceData,
     options: {
        profile: {
	   name: identity.name || identity.username || identity.nickname || identity.CharacterName || identity.userName || identity.preferred_username || (identity.user && identity.user.name)
	}
     }
};
```

**Comment by torgeirl** [comment]
@ToniIltanen: I started looking at the Custom OAuth implementation for another problem, but realized the current build just contains a bunch of hotfixes for some services (see a few lines above `const data = {` in [custom_oauth_server.js](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-custom-oauth/server/custom_oauth_server.js)) and solved my problem by extending a hotfix with PR #8627. A more elegant way for admins to provide custom mapping for OAuth fields sounds much better than the hotfixes, and something I also could help with.

Another thing to consider is whether the hotfixed custom oauths even should be custom, and not just made into regular oauth settings as the fixes aren't visual to anyone not diving into the code.

**Comment by torgeirl** [comment]
I've now looked a little into this issue, and it's incorrect that the email field isn't used.

The picture field is a different story. PR #5338 introduced Set Default Avatar with the help text "Tries to determine default avatar based on OAuth Account or Gravatar", but the OAuth part of it seems just to be intent. For some reason, the handling of default avatars was added to setUsername.js over setUserAvatar.js and it doesn't seem to be any handling of default avatars for OAuth, at least not in a general way that would also benefit custom OAuths.

I'm unsure where this should be added as we would want identity.picture that often isn't a full URL, but a value that must be used in relation to the service that provided the profile handled differently for each service, and I don't see exactly how we can implement that in the current code.
